src/main/java/randoop/contract/ObserverEqArray.java

```diff
@@ -0,0 +1,211 @@
+package randoop.contract;
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+import java.util.StringJoiner;
+import org.apache.commons.lang3.ArrayUtils;
+import org.junit.Assert;
+import org.plumelib.util.StringsPlume;
+import randoop.main.RandoopBug;
+import randoop.reflection.AccessibilityPredicate;
+import randoop.sequence.Value;
+import randoop.types.JavaTypes;
+import randoop.types.PrimitiveTypes;
+import randoop.types.TypeTuple;
+
+/**
+ * A check recording the current state of an array during execution.
+ *
+ * <p>ObserverEqArray checks are not checks that must hold of all objects of a given class (unlike a
+ * check like {@link EqualsReflexive}, which must hold for any objects, no matter its execution
+ * context).
+ */
+public final class ObserverEqArray extends ObjectContract {
+
+  /** The run-time result of calling the observer: an array of literals. */
+  private Object value;
+
+  /**
+   * The maximum delta between the expected and actual for which both numbers (doubles or floats)
+   * are still considered equal.
+   */
+  private static final double FLOATING_POINT_DELTA = 1e-15;
+
+  /**
+   * Creates a new ObserverEqArray.
+   *
+   * @param value the run-time result of calling the observer: an array of literals
+   * @param isAccessible the accessibility predicate
+   */
+  public ObserverEqArray(Object value, AccessibilityPredicate isAccessible) {
+    this.value = value;
+    if (!isLiteralType(value.getClass().getComponentType(), isAccessible)) {
+      throw new RandoopBug(
+          String.format(
+              "Cannot represent %s as a literal",
+              StringsPlume.toStringAndClass(value.getClass().getComponentType())));
+    }
+  }
+
+  /**
+   * Returns true if values of the class can be represented as literals in Java source code.
+   *
+   * @param cls a class
+   * @param isAccessible the accessibility predicate
+   * @return true iff the class is a primitive, boxed primitive, String, Class, or accessible Enum
+   */
+  public static boolean isLiteralType(Class<?> cls, AccessibilityPredicate isAccessible) {
+    if (cls == Class.class || cls == String.class) {
```

Comment:
Should the `Class.class` case also be associated with an `isAccessible()` test, as the `isEnum()` case is?  (I don't know the answer to this, but if the answer is "no", please document that fact in a comment.

Reply:
Not sure if this answers your question, but isn't Class.class always accessible? (built-in)